src/Translation.h

```diff
@@ -0,0 +1,73 @@
+#ifndef TRANSLATION_H
+#define TRANSLATION_H
+#include <QMetaType>
+#include <QString>
+#include <memory>
+
+namespace marian {
+    namespace bergamot {
+        class Response;
+    }
+}
+
+/**
+ * Word alignment information, with information about a span in the translation
+ * output as answer to querying which parts 'belong' to a span in the
+ * translation input.
+ */
+struct WordAlignment {
+    std::size_t begin; // Note: char offset in translation (not byte offset)
+    std::size_t end;
+    float prob; // Range: 0.0 - 1.0
+};
+
+/**
+ * Wrapper around a translation response from the bergamot service. Hides that
+ * interface from the rest of the Qt code, and provides utility functions to
+ * access alignment information with character offsets instead of byte offsets.
+ */
+class Translation {
+private:
+    // Note: I would have liked unique_ptr, but that does not go well with
+    // passing Translation objects through Qt signals/slots.
+    std::shared_ptr<marian::bergamot::Response> response_;
+
+    int speed_;
```

Comment:
The comment with regards what `speed_` is got lost with the code re-organisation.